lib/src/geometry/collision_detection.dart

```diff
@@ -1,18 +1,32 @@
 import '../../extensions.dart';
 import '../components/mixins/collidable.dart';
 
+void aloneCollisionDetection(Collidable collidableX, Collidable collidableY) {
+  final points = intersections(collidableX, collidableY);
+  if (points.isNotEmpty) {
+    collidableX.onCollision(points, collidableY);
+    collidableY.onCollision(points, collidableX);
+  }
+}
+
 /// Check whether any [Collidable] in [collidables] collide with each other and
 /// call their onCollision methods accordingly.
-void collisionDetection(List<Collidable> collidables) {
-  for (var x = 0; x < collidables.length; x++) {
-    final collidableX = collidables[x];
-    for (var y = x + 1; y < collidables.length; y++) {
-      final collidableY = collidables[y];
-      final points = intersections(collidableX, collidableY);
-      if (points.isNotEmpty) {
-        collidableX.onCollision(points, collidableY);
-        collidableY.onCollision(points, collidableX);
-      }
+void collisionsDetection(
+  List<Collidable> activeCollidables,
+  List<Collidable> noActiveCollidables,
+) {
+  for (var x = 0; x < activeCollidables.length; x++) {
+    /// The property [activecollidables] of [collidablex] is  true
+    final collidableX = activeCollidables[x];
+
+    /// Detection [activeCollidables]
+    for (var y = x + 1; y < activeCollidables.length; y++) {
+      aloneCollisionDetection(collidableX, activeCollidables[y]);
+    }
+
+    /// Detection [noActiveCollidables]
+    for (var y = 0; y < noActiveCollidables.length; y++) {
```

Comment:
This can use the `for in` loop notation.

Reply:
@spydon 
OK, I'll try to make the [active] API more usable. I agree with you, but I still have doubts about how to use [oncollision] simply. For example, when a collisionable box wants to climb over the obstacles made up of other collisionable boxes, I have doubts about this callback method. I hope to improve these interfaces so that they can be used effectively, I'd like to have a discussion. I've seen it

https://github.com/flame-engine/flame/blob/master/doc/collision_ detection.md

I think it needs some knowledge of mathematics. At the same time, I'm sorry for my lack of knowledge and experience in 2D game engine. Thank you for your attention to my news release